### PR TITLE
New post: Simple Rust, Part Three

### PR DESCRIPTION
**Pull request by IanWhitney**

### CODE REVIEW

File: _posts/2015-11-27-simple-rust-part-three.markdown [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+
+Let's start with `Vec`. In the [last post](http://designisrefactoring.com/2015/11/13/simple-rust-part-two/) we looked at Array, which suited our needs because it is immutable. I mentioned that there were also mutable arrays, and here they are -- Vectors, or Vec for short.
+
+Remember that when we write an array like this:
+
+{% highlight rust %}
+let an_array = ["tan", "stand", "at"];
+{% endhighlight %}
+
+it doesn't have the type of `Array`. Instead its type is `[&str,3]`, the combination of the type and number of its elements.
```
Comment (Dr-Emann) [inline _posts/2015-11-27-simple-rust-part-three.markdown]: Array types use `;` to separate the type of items, and the length: The type should be `[&str;3]`, or, more pedantically, `[&'static str; 3]`

Comment (IanWhitney) [inline _posts/2015-11-27-simple-rust-part-three.markdown]: Thanks for pointing that out!

**Comment by cafreeman** [comment]
Unless I'm missing something, I believe you have to declare `anagrams_for` as a public function before you can get the tests to pass. For instance, I wasn't able to pass the first test until my function looked like this:

```
pub fn anagrams_for<'a>(source: &str, inputs: &[&'a str]) -> Vec<&'a str> {
    vec![]
}
```

Did you experience the same thing? If so, it might be worth updating the code example to include the `pub` declaration.

**Comment by IanWhitney** [comment]
You are right, @cafreeman. In a lot of cases I was writing code as if everything was in a single file, like [this](https://play.rust-lang.org/?gist=dfe56bd24bc2ca6a9466&version=stable). In that case, I can leave the `pub` off. But when you're dealing with the Exercism code, you have to have `pub` since you're importing your anagram code as an external crate. I'll try to make that more clear.

**Pull request merged**